x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/job/MetricConfig.java

```diff
@@ -108,18 +109,26 @@ public void validateMappings(Map<String, Map<String, FieldCapabilities>> fieldCa
         Map<String, FieldCapabilities> fieldCaps = fieldCapsResponse.get(field);
         if (fieldCaps != null && fieldCaps.isEmpty() == false) {
             fieldCaps.forEach((key, value) -> {
-                if (RollupField.NUMERIC_FIELD_MAPPER_TYPES.contains(key)) {
+                if (RollupField.NUMERIC_FIELD_MAPPER_TYPES.contains(key) || RollupField.DATE_FIELD_MAPPER_TYPE.equals(key)) {
                     if (value.isAggregatable() == false) {
                         validationException.addValidationError("The field [" + field + "] must be aggregatable across all indices, " +
                                 "but is not.");
                     }
+                    if (RollupField.DATE_FIELD_MAPPER_TYPE.equals(key) &&
+                        RollupField.SUPPORTED_DATE_METRICS.containsAll(metrics) == false) {
+                        List<String> unsupportedMetrics = new ArrayList<>(metrics);
+                        unsupportedMetrics.removeAll(RollupField.SUPPORTED_DATE_METRICS);
+                        validationException.addValidationError("Only the metrics " + RollupField.SUPPORTED_DATE_METRICS.toString() +
```

Comment:
Super tiny nit: could we wrap the `SUPPORTED_DATE_METRICS` with brackets (`[...]`)?  Makes it easier for the user to see the difference in message and dynamic values.

Ditto to below with `unsupportedMetrics`

Reply:
@polyfractal A `Collection<String>` object when transformed to a `String` transforms to a `[Metric1, Metric2]`. To clarify, you want me to add another bracket so that the printout will be `[[Metric1, Metric2]]` ?

Reference Repl: https://repl.it/@ben_w_trent/QuerulousSubtleApplicationframework